spec.md

```diff
@@ -542,6 +542,172 @@ The following string-encoded integer literals all have the value `-42`:
 Radix-specified integer literals are only usable when constructing hardware
 integer literals.  Any use in place of an integer is disallowed.
 
+# Circuit Components
+
+Circuit components are the named parts of a module corresponding to physical hardware.
+
+There are seven **kinds** of circuit components.
+They are: nodes, wires, registers, output ports, input ports, submodule instances, and memories.
+
+Circuit components can be connected together in the hardware (see [@sec:connects]).
+
+Each circuit component in a module is associated
+with a type called its type ([@sec:types]).
+This is the type of data that travels across the component.
+It is used to determine the legality of connections.
+
+
+## Nodes
+
+Nodes are named expressions in FIRRTL.
+
+Example:
+
+```firrtl
+    node mynode = and(in, UInt<4>(1))
+```
```

Comment:
Nit: Can the PR use a consistent indenting of these examples? I tend to always left align them.

Ditto throughout.